src/model.d.ts

```diff
@@ -2208,17 +2230,28 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
    * @param {string} alias
    * @return {boolean}
    */
+  // TODO: rename to hasAssociation
```

Comment:
I would suggest doing it during the migration of model.js to TypeScript. Doesn't really matter if we do it before or after the release of v7, we'll probably deprecate `hasAlias` and add `hasAssociation` to avoid another breaking change